dafni_cli/model/model.py

```diff
@@ -1,262 +1,348 @@
-import datetime as dt
+from dataclasses import dataclass
+from datetime import date, datetime
+from typing import ClassVar, List, Optional
 
 import click
-from dateutil import parser
 
-from dafni_cli.api.models_api import get_model
-from dafni_cli.api.session import DAFNISession
-from dafni_cli.auth import Auth
-from dafni_cli.consts import CONSOLE_WIDTH, TAB_SPACE
-from dafni_cli.model.model_metadata import ModelMetadata
-from dafni_cli.utils import print_json, prose_print
+from dafni_cli.api.auth import Auth
+from dafni_cli.api.parser import ParserBaseObject, ParserParam, parse_datetime
+from dafni_cli.consts import (
+    CONSOLE_WIDTH,
+    TAB_SPACE,
+)
+from dafni_cli.model.inputs import ModelInputs
+from dafni_cli.model.outputs import ModelOutputs
+from dafni_cli.utils import prose_print
 
 
-class Model:
+@dataclass
+class ModelMetadata(ParserBaseObject):
+    """Dataclass representing a DAFNI model's metadata
+
+    Attributes:
+        display_name (str): The display name of the Model
+        name (str): Name of the model
+        summary (str): A short summary of the Model's function
+        status (str): Status of model ingest
+            P - Pending
+            F - Failed
+            L - Live
+            S - Superseded
+            D - Deprecated
+
+        The following are only present for the /model/<version_id> endpoint
+        (but are guaranteed for it)
+        --------
+        description (str): A rich description of the Model's function
+        publisher (str): The name of the person or organisation who has
+                         published the Model
+        source_code (str): A URL pointing to the source code for this Model
```

Comment:
Is source_code definitely guaranteed?

Reply:
From what I can see in NIMS it will exist but just be an empty string if it hasn't been supplied which is what I meant by guaranteed in this case - it will exist in the response, but its value may not be given. I will slightly amend the docstring to make it clearer.